tests/lwip_sock_udp/stack.c

```diff
@@ -213,7 +213,7 @@ void _prepare_send_checks(void)
         struct nd6_neighbor_cache_entry *nc = &neighbor_cache[i];
         if (nc->state == ND6_NO_ENTRY) {
             nc->state = ND6_REACHABLE;
-            memcpy(&nc->next_hop_address, remote6, sizeof(ip6_addr_t));
+            memcpy(&nc->next_hop_address, remote6, sizeof(remote6));
```

Comment:
```suggestion
            memcpy(&nc->next_hop_address, remote6, sizeof(nc->next_hop_address));
```

Reply:
See https://github.com/aabadie/RIOT/pull/14#discussion_r336621786

Reply:
So [`ip6_addr_t`](https://github.com/yarrick/lwip/blob/master/src/include/lwip/ip6_addr.h#L59) contains an additional `zone` field.

```suggestion
            memcpy(&nc->next_hop_address.addr, remote6, sizeof(nc->next_hop_address.addr));
```

should work.

Reply:
Arghs should have come to that conclusion myself. @aabadie please apply @benpicco's change suggestion. However, the test should then be re-run at least for `native`. I remember that lwIP was somewhat finicky about that zone field when it comes to neighbor discovery.

Reply:
> However, the test should then be re-run at least for native.

Huh? I'm pretty sure `sizeof(nc->next_hop_address.addr) == sizeof(remote6)` and `&nc->next_hop_address.addr == &&nc->next_hop_address`.

It still copies 16 bytes to the same address - it just ensures that those 16 bytes are derived from the destination buffer instead of the source buffer and that the target is not the struct itself, but the first member of the struct at offset 0.

For the generated code, this makes no difference.